pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -101,8 +90,10 @@ func (h *handler) getArgsEnvAndStatus(infra *infraObject, args map[string]interf
 		},
 		Data: getWhitelistedEnvVars(),
 	}
-	machine, err := rke2.GetMachineByOwner(h.machineCache, infra.meta)
-	if err != nil && (create || !errors.Is(err, rke2.ErrNoMachineOwnerRef)) {
+	// the owner reference may be deleted if the machine is cleaned up forcefully
+	machineName := infra.meta.GetLabels()[CapiMachineName]
```

Comment:
Potentially, do we want to "fallback" one way or the other to the owner ref or the label if one or the other doesn't exist? Not an actual problem, just wondering.

Reply:
We could, but if the label is not present, we never reconciled far enough to create infra anyways, so we don't have to.